frontend-react/src/hooks/UsePagination.test.ts

```diff
@@ -0,0 +1,474 @@
+    numRecords: number,
+    startCursor = 1
+): SampleRecord[] {
+    return range(startCursor, startCursor + numRecords).map(
+        (c) => ({ cursor: c.toString() } as SampleRecord)
+    );
+}
+
+describe("getSlots", () => {
+    test("when the set is unbounded", () => {
+        expect(getSlots(1)).toStrictEqual([
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            OVERFLOW_INDICATOR,
+        ]);
+        expect(getSlots(2)).toStrictEqual([
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            OVERFLOW_INDICATOR,
+        ]);
+        expect(getSlots(3)).toStrictEqual([
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            OVERFLOW_INDICATOR,
+        ]);
+        expect(getSlots(4)).toStrictEqual([
+            1,
+            2,
+            3,
+            4,
+            5,
+            6,
+            OVERFLOW_INDICATOR,
+        ]);
+        expect(getSlots(5)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            4,
+            5,
+            6,
+            7,
+            OVERFLOW_INDICATOR,
+        ]);
+        expect(getSlots(20)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            19,
+            20,
+            21,
+            22,
+            OVERFLOW_INDICATOR,
+        ]);
+    });
+
+    test("when the set is bounded", () => {
+        expect(getSlots(2, 8)).toStrictEqual([
+            1,
+            2,
+            3,
+            4,
+            5,
+            OVERFLOW_INDICATOR,
+            8,
+        ]);
+        expect(getSlots(4, 10)).toStrictEqual([
+            1,
+            2,
+            3,
+            4,
+            5,
+            OVERFLOW_INDICATOR,
+            10,
+        ]);
+        expect(getSlots(6, 10)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            5,
+            6,
+            7,
+            OVERFLOW_INDICATOR,
+            10,
+        ]);
+        expect(getSlots(7, 10)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            6,
+            7,
+            8,
+            9,
+            10,
+        ]);
+        expect(getSlots(8, 10)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            6,
+            7,
+            8,
+            9,
+            10,
+        ]);
+        expect(getSlots(9, 10)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            6,
+            7,
+            8,
+            9,
+            10,
+        ]);
+        expect(getSlots(10, 10)).toStrictEqual([
+            1,
+            OVERFLOW_INDICATOR,
+            6,
+            7,
+            8,
+            9,
+            10,
+        ]);
+    });
+
+    test("when there are no more than seven pages", () => {
+        expect(getSlots(1, 1)).toStrictEqual([1]);
+
```

Comment:
Fantastic! This is something I wondered about after we took a peek during sprint review. Glad to know we have it tested, too 👍🏼